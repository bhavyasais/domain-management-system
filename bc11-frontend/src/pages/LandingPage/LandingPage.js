import React from 'react';
import SideNavigationBar from '../../components/molecules/SideNavigationBar/SideNavigationBar';
import Topbar from '../../components/organisms/Topbar/Topbar';
import AppBanner from '../../components/molecules/AppBanner/AppBanner';
import { makeStyles } from '@material-ui/core/styles';
import LandingPageTemplate from '../../components/templates/LandingPageTemplate/LandingPageTemplate';
import {useDispatch } from 'react-redux';
import { updateUser } from '../../redux/actions/userActions';
const useStyles = makeStyles(() => ({
  root: {
    marginTop: '229px',
  },
  topmargin: {
    marginTop: '-10px',
    fontSize: '30px',
    color: '#666666',
  },
  buttonmargin: {
    marginTop: '10px',
    width: '206px',
    height: '44px',
    backgroundColor: '#1d4cd7',
    fontFamily: 'Roboto',
    fontSize: '18px',
    fontWeight: 'normal',
    fontStretch: 'normal',
    fontStyle: 'normal',
    lineHeight: 'normal',
    letterSpacing: 'normal',
  },
  alignCenter: {
    maxHeight: '310px',
    maxWidth: '300px',
  },
}));

export default function LandingPage(props) {
  const classes = useStyles();
  const dispatch = useDispatch(); //updating user details from redux store
  React.useEffect(() => {
    dispatch(updateUser({ userName: 'Bhavya' }));
  }, []);
  const options = {
    sidebar: <SideNavigationBar />,
    topbar: <Topbar label="Centralized  Communication Command Center" />,
    banner: <AppBanner height={300} width={310} childClasses={classes} />,
  };
  return <LandingPageTemplate childComponents={options} />;
}
