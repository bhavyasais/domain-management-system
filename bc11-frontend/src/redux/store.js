import {createStore, combineReducers } from "redux";
import {userReducer} from "./reducers/userReducer";

const initialState = {};

const store = createStore(
  combineReducers({ user: userReducer}),
  initialState,
);
export default store;
