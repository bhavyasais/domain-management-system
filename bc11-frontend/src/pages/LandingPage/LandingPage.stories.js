import React from "react";
import { storiesOf } from "@storybook/react";
import LandingPage from "./LandingPage";

storiesOf("Templates/Landing Page", module).add("Landing Page", () => (
  <LandingPage label="Centralized  Communication Command Center" />
));
