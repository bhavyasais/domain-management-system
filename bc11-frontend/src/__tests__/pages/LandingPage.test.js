import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import LandingPage from '../../pages/LandingPage/LandingPage';
import store from '../../redux/store';

describe('CollabModal', () => {
  it('should render correctly', () => {
    const wrapper = renderer.create(
      <Provider store={store}>
        <LandingPage />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
