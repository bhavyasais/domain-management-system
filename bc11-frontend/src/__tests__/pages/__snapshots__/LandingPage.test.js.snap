// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollabModal should render correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container"
>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <div
      className="MuiGrid-root SideNavigationBar-root-5 MuiGrid-container MuiGrid-direction-xs-column"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <p
          className="MuiTypography-root SideNavigationBar-header-9 MuiTypography-body1"
        >
          io
        </p>
      </div>
      <div
        className="MuiGrid-root SideNavigationBar-imageContainer-7 MuiGrid-container MuiGrid-item"
      >
        <img
          className="SideNavigationBar-image-6"
          onClick={[Function]}
          src="atoms_icon_pie.svg"
        />
      </div>
      <div
        className="MuiGrid-root SideNavigationBar-imageContainer-7 SideNavigationBar-activeIcon-8 MuiGrid-container MuiGrid-item"
      >
        <img
          className="SideNavigationBar-image-6"
          onClick={[Function]}
          src="atoms_icon_graph.svg"
        />
      </div>
      <div
        className="MuiGrid-root SideNavigationBar-imageContainer-7 MuiGrid-container MuiGrid-item"
      >
        <img
          className="SideNavigationBar-image-6"
          onClick={[Function]}
          src="mail.png"
        />
      </div>
      <div
        className="MuiGrid-root SideNavigationBar-imageContainer-7 MuiGrid-container MuiGrid-item"
      >
        <img
          className="SideNavigationBar-image-6"
          onClick={[Function]}
          src="atoms_icon_excel.svg"
        />
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
    style={
      Object {
        "width": "95.5%",
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="MuiGrid-root makeStyles-container-10 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
        >
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <p
              className="MuiTypography-root makeStyles-color-11 MuiTypography-body1"
            >
              Centralized  Communication Command Center
            </p>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
            >
              <div
                className="MuiGrid-root makeStyles-marginRight-12 MuiGrid-item"
              >
                <div>
                  <button
                    className="MuiButtonBase-root MuiIconButton-root"
                    disabled={false}
                    onBlur={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="button"
                  >
                    <span
                      className="MuiIconButton-label"
                    >
                      <span
                        className="MuiBadge-root"
                        icon={<Memo(TextsmsOutlinedIcon) />}
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M20 2H4c-1.1 0-2 .9-2 2v18l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm0 14H6l-2 2V4h16v12zM7 9h2v2H7zm4 0h2v2h-2zm4 0h2v2h-2z"
                          />
                        </svg>
                        <span
                          className="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorSecondary MuiBadge-dot"
                        >
                          
                        </span>
                      </span>
                    </span>
                  </button>
                </div>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item"
              >
                <div>
                  <button
                    className="MuiButtonBase-root MuiIconButton-root"
                    disabled={false}
                    onBlur={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="button"
                  >
                    <span
                      className="MuiIconButton-label"
                    >
                      <span
                        className="MuiBadge-root"
                        icon={<Memo(NotificationsOutlinedIcon) />}
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.9 2 2 2zm6-6v-5c0-3.07-1.63-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.64 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2zm-2 1H8v-6c0-2.48 1.51-4.5 4-4.5s4 2.02 4 4.5v6z"
                          />
                        </svg>
                        <span
                          className="MuiBadge-badge MuiBadge-anchorOriginTopRightCircle MuiBadge-colorSecondary MuiBadge-dot"
                        >
                          
                        </span>
                      </span>
                    </span>
                  </button>
                </div>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item"
              >
                <button
                  aria-haspopup="true"
                  className="MuiButtonBase-root MuiButton-root MuiButton-disableElevation"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiButton-label"
                  >
                    <div
                      className="MuiAvatar-root MuiAvatar-circle makeStyles-avatarSize-13"
                    >
                      <img
                        className="MuiAvatar-img"
                        src="https://cdn.zeplin.io/5fa102170abc4fbe7467eaa5/screens/438d5cb7-5113-4c26-9a9f-109b56acf94c.png"
                      />
                    </div>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="makeStyles-root-1"
        >
          <div
            align="center"
          >
            <div
              style={
                Object {
                  "backgroundColor": undefined,
                  "height": 300,
                  "position": "relative",
                  "width": 310,
                }
              }
            >
              <img
                onError={[Function]}
                onLoad={[Function]}
                src="banner.svg"
                style={
                  Object {
                    "display": "block",
                    "height": 0,
                    "left": 155,
                    "position": "absolute",
                    "top": 150,
                    "width": 0,
                  }
                }
              />
            </div>
          </div>
          <h5
            className="MuiTypography-root makeStyles-topmargin-2 MuiTypography-h5 MuiTypography-alignCenter"
          >
            Your active permissions will appear here.
            <div>
              <div>
                <button
                  className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-buttonmargin-3"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#1d4cd7",
                      "maxWidth": undefined,
                    }
                  }
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiButton-label makeStyles-buttonLabel-17"
                  >
                    <p
                      className="MuiTypography-root makeStyles-buttonLabel-17 MuiTypography-body1"
                    >
                      Enable Access
                    </p>
                  </span>
                </button>
              </div>
              <div
                className="makeStyles-root-18"
              />
            </div>
          </h5>
        </div>
      </div>
    </div>
  </div>
</div>
`;
